metabase/metabase
# Update Google Sign In

Fixes #17429

Please see issue above for justification and details.

tl;dr: New Metabase users logging in with Google Sign In may not be able to do so until we upgrade Google's Sign In library.

🎗️ 
There's more to do here, but this PR should do the job if we decide to ship very soon, or be able to test with users very soon.
In my tests it's been working fine.

### Advantages of this solution

Implementation is really straightforward.
We may be able to clean up a lot of code and remove a lot of indirection if we use this.

We may later quickly switch to the "one tap" version of the button, which should make signing in faster: button will suggest an email account to use, should the user have more than one Google emails. Plus, developers can add tips on what that email should be.

### Trade-offs to note

1. Button is now the official Google button, not a customized button.
We can't do drop shadows or vertical padding, for example.
Button look can be customized under Google's parameters.

2. Some unit tests have been skipped for agility, and can be rethought if we go with this solution.

3. Metabase users with languages other than English will see a flash of "Sign in With Google" before they see the same copy in their language.

## How to Test

1. Go to http://localhost:3000/admin/settings/authentication/google
2. Add Google Client ID [you can find here](https://metaboat.slack.com/archives/C03E3Q5T4P4/p1653085210635789)
3. Log out
4. Log in with Google 

You should be able to log in.
